Source/HttpMultipartParser/MultipartFormDataParser.cs

```diff
@@ -357,9 +384,38 @@ public static MultipartFormDataParser Parse(Stream stream, Encoding encoding, in
         ///     A new instance of the <see cref="MultipartFormDataParser"/> class.
         /// </returns>
         public static MultipartFormDataParser Parse(Stream stream, string boundary, Encoding encoding, int binaryBufferSize)
+        {
+            return MultipartFormDataParser.Parse(stream, null, encoding, binaryBufferSize, null);

```

Comment:
you forgot to pass `boundary` when invoking `MultipartFormDataParser.Parse`.

```suggestion
            return MultipartFormDataParser.Parse(stream, boundary, encoding, binaryBufferSize, null);
```

Reply:
Sry for that. Was under high time pressure yesterday. Needed this fix for an important project and started the pull request a bit early.